DataFormats/Common/interface/DetSet.h

```diff
@@ -66,6 +66,8 @@ namespace edm {
     const_reference operator[](size_type i) const { return data[ i ]; }
     void reserve(size_t s) { data.reserve(s); }
     void push_back(const T & t) { data.push_back(t); }
+    template<class... Args>
+    auto emplace_back(Args&&... args) { return data.emplace_back(std::forward<Args>(args)...); }
```

Comment:
I didid a better job of researching the problem and found the actual fix is
```suggestion
    decltype(auto) emplace_back(Args&&... args) { return data.emplace_back(std::forward<Args>(args)...); }
```
That will make the return type a reference, which is what we want.

Reply:
See https://en.cppreference.com/w/cpp/language/auto for details.

Reply:
Indeed, I was not yet aware of this `decltype(auto)`... this looks way more succinct than a `__gnu_cxx::__alloc_traits<_Vector_base<T, std::allocator<T> >::__gnu_cxx::__alloc_traits<_Alloc>::template
  rebind<T>::other>::reference`. Many thanks! It seems that you did all the job there :)